db/models.py

```diff
@@ -46,3 +52,62 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class User(AbstractUser):
+    pass
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(to=User, on_delete=models.CASCADE)
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return f"{self.created_at}"
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        to=MovieSession, on_delete=models.CASCADE
+    )
+    order = models.ForeignKey(to=Order, on_delete=models.CASCADE)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                fields=["row", "seat", "movie_session"],
+                name="unique_ticket_on_specified_movie_session"
+            )
+        ]
+
+    def __str__(self) -> str:
+        return f"{self.movie_session.movie.title}" \
+               f" {self.movie_session.show_time} " \
+               f"(row: {self.row}, seat: {self.seat})"
+
+    def clean(self) -> None:
+        if not (1 <= self.seat <= self.movie_session.cinema_hall.seats_in_row):
+            raise ValidationError({
+                "seat": f"seat number must be in available range: "
+                        f"(1, seats_in_row):"
+                        f" (1, {self.movie_session.cinema_hall.seats_in_row})"
+            })
+        if not (1 <= self.row <= self.movie_session.cinema_hall.rows):
+            raise ValidationError({
+                "row": f"row number must be in available range: (1, rows):"
+                       f" (1, {self.movie_session.cinema_hall.rows})"
+            })
+
+    def save(
+            self, force_insert: bool = False, force_update: bool = False,
+            using: bool = None, update_fields: bool = None
```

Comment:
typehint:
bool | None = None

Reply:
redone

Reply:
Not done
if default argument is None
then type is always Optional[and type of your variable]

Reply:
@phaishuk im on the way, im marking that i reviewed this comment)

Reply:
@ArsenPidhoretskyi  im on the way, im marking that i reviewed this comment)